packages/compat/src/signed-versioned-transaction.ts

```diff
@@ -0,0 +1,37 @@
+import { VersionedTransaction } from '@solana/web3.js';
+
+type NonEmptyArray<T> = [T, ...T[]];
+
+export type SignedVersionedTransaction = Omit<VersionedTransaction, 'signatures'> & {
+    signatures: NonEmptyArray<Uint8Array>;
+};
+
+function signatureIsAllZeros(signature: Uint8Array) {
+    return signature.every(byte => byte === 0);
+}
+
+export function isSignedVersionedTransaction(
+    versionedTransaction: VersionedTransaction
+): versionedTransaction is SignedVersionedTransaction {
+    return versionedTransaction.signatures.findIndex(sig => !signatureIsAllZeros(sig)) > -1;
+}
```

Comment:
This isn't really compat-y. This is an operation over exclusively the legacy data type. It probably doesn't belong here.

Instead, I'd do one of:

1. Put this in the legacy library, if it's useful.
2. _Let_ them convert it to modern _first_, and then use the modern assertion `assertIsFullySignedTransaction` (which we have yet to write in `@solana/transactions`).

Reply:
I don't think `IFullySignedTransaction` is what we want here, you should be able to convert a transaction regardless of if it's signed (eg for simulation I want it to be unsigned/partially signed)

But you're right that we could do the assertion on the modern transaction, just to `ITransactionWithSignatures`. So we just do `fromVersionedTransactionWithBlockhash`, that gives you the transaction but not `ITransactionWithSignatures`, and you can then assert it has signatures  or cast to get `ITransactionWithSignatures` if you care about that.

The remaining question though is what to do with those all-zero signatures when I convert them? The new transaction model (AFAICT) just stores actual signatures in a dictionary. We could remove the filter so when you pass a transaction it ends up with the all-zero signature in that dictionary. Presumably that'd work fine, it's just not what you'd really expect with the new transaction model IMO.